fuse/utils/multiprocessing/run_multiprocessed.py

```diff
@@ -260,15 +277,31 @@ def inner(f: callable):
         def wrapper(*args, **kwargs):
             # create the machinery python uses to fork a subprocess
             # and run a function in it.
-            p = mp.Process(target=f, args=args, kwargs=kwargs)
+            # p = mp.Process(target=f, args=args, kwargs=kwargs)
+            p = Process(target=f, args=args, kwargs=kwargs) # using a subclass of Process
             p.start()
             try:
                 p.join(timeout=timeout)
             except:
                 p.terminate()
                 raise
-            
+
+            if p.exception:
+               error, traceback = p.exception
+               print(f"process func {f} had an exception: {error}")
+               print(traceback)
+               raise RuntimeError(f"process func {f} had an exception: {error}")
```

Comment:
great !!

Reply:
All rights belong to @michalozeryflato :) (I copied it with minor modifications)